framework/src/variables/MooseVariableFV.C

```diff
@@ -463,7 +463,8 @@ MooseVariableFV<OutputType>::getElemValue(const Elem * const elem, const StateAr
   // while the libMesh solution is frozen in the non-perturbed state
   const auto & global_soln = (state.state == 0)
                                  ? *this->_sys.currentSolution()
-                                 : this->_sys.solutionState(state.state, state.iteration_type);
+                                 : static_cast<const SystemBase &>(this->_sys)
```

Comment:
Why is this cast necessary?

Reply:
To use the const version of that routine. The non const is not thread safe

Reply:
Do a const cast then and not a static one

Reply:
`std::as_const(this->_sys)`

Reply:
beautiful code suggestion. First time I’ve seen that!

Reply:
Also I feel like a better fix would be to make the non-const method thread-safe and then you do not have to do any casts here nor will other developers who attempt to use this method not knowing the non-const version is not thread safe

Reply:
So add a mutex or something

Reply:
I added an assert to forbid requesting previous states on the fly in threaded regions